### PR TITLE
Unify trace and metric exporter helpers

### LINKED ISSUE #800: Inconsistency between trace and metric exporter helpers
The metric exporters include functions named: "NewRawExporter" (except prometheus, since it doesn't implement the Exporter interface), "NewExportPipeline", and "InstallNewPipeline".

The Jaeger exporter has "NewRawExporter" and "NewExportPipeline" but not "InstallNewPipeline", instead it uses a boolean `RegisterGlobal` to indicate whether the pipeline should be installed as the global. 

The Zipkin exporter has none of these functions.

I would prefer if we added InstallNewPipeline to Jaeger and all three of these functions to Zipkin, to bring about consistency. This came up in #796 because the JAEGER_DISABLED environment variable should probably cause "InstallExportPipeline" to install a No-op SDK, not to install a real SDK with a disabled exporter.

### PR DESCRIPTION
**Pull request by matej-g**
Brings consistency to helper methods of trace and metric exporters (Prometheus, Jaeger, Zipkin). Each of these now has 3 helper methods with the same purpose:
 - `NewRawExporter` - method to return exporter only, with default or passed options
- `NewExportPipeline` - method which creates a new exporter and uses it to return a provider with recommended configuration
- `InstallNewPipeline` - same as `NewExportPipeline`, but it directly registers provider as global

Specifically the changeset includes:
- for Jager exporter, introduced `InstallNewPipeline` and removed `RegisterGlobal` option instead
- for Zipkin exporter, introduced pipeline methods and adjusted the new exporter method 

New tests, example updates and minor improvements (e.g. added documentation) are included here as well.

### Related Issues:
Resolves #800

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - OTLP .proto files from `open-telemetry/opentelemetry-proto` imported as a git submodule under `internal/opentelemetry-proto`.
    References to `github.com/open-telemetry/opentelemetry-proto` changed to `go.opentelemetry.io/otel/internal/opentelemetry-proto-gen`. (#948)
 
+### Changed
```
Comment (MrAlias) [inline CHANGELOG.md:17]: Can we break out the specifics of this change set from the end user perspective. These notes are commonly used when upgrading or evaluating releases so it will be good to make it clear the things that were removed or replace.

I.e. 

> Specifically the changeset includes:
> * for Jager exporter, introduced InstallNewPipeline and removed RegisterGlobal option instead
> * for Zipkin exporter, introduced pipeline methods and adjusted the new exporter method

File: exporters/trace/jaeger/jaeger_test.go [hunk 2]
```diff
@@ -37,26 +37,34 @@ import (
...
-		RegisterAsGlobal(),
 	)
 	defer fn()
 	assert.NoError(t, err)
-	assert.Same(t, tp, global.TraceProvider())
+	assert.IsType(t, &sdktrace.Provider{}, global.TraceProvider())
 }
 
-func TestNewExporterPipelineWithoutRegistration(t *testing.T) {
+func TestInstallNewPipeline_WithDisabled(t *testing.T) {
```
Comment (MrAlias) [inline exporters/trace/jaeger/jaeger_test.go:49]: I'm in favor of sticking with Go conventions of mixed capitalization naming and not introducing underscores.

If we want to delineate the testing it might be useful to abstract the "`TestInstallNewPipeline`" part of the test from the different cases and use `t.Run` with separate broken out functions.

File: exporters/trace/zipkin/zipkin.go [hunk 3]
```diff
@@ -35,58 +40,116 @@ type Exporter struct {
...
+
+	o := options{}
+	for _, opt := range opts {
+		opt(&o)
 	}
-	opts := Options{}
-	for _, o := range os {
-		o(&opts)
+	if o.serviceName == "" {
+		o.serviceName = defaultServiceName
```
Comment (MrAlias) [inline exporters/trace/zipkin/zipkin.go:107]: Having a default service name doesn't seem correct. The zipkin model need a unique service name to delineate services in its representation of the distributed trace. With a design that defaults to a common service name it seems like it will lead to a class of error and frustration for the user where a "successful" setup and run of the system will contain invalid data.

It might make sense to have an exception here to the similar function signatures across the exporters. Given Zipkin has different requirements, having the service name be an argument would make sense to me.

File: exporters/trace/zipkin/zipkin_test.go [hunk 4]
```diff
@@ -27,13 +27,97 @@ import (
...
+
+func TestNewExportPipeline(t *testing.T) {
+	tp, err := NewExportPipeline(
+		"http://localhost:9411/api/v2/spans",
+	)
+	assert.NoError(t, err)
+	assert.NotEqual(t, tp, global.TraceProvider())
+}
+
+func TestNewExportPipeline_WithSDK(t *testing.T) {
```
Comment (MrAlias) [inline exporters/trace/zipkin/zipkin_test.go:56]: Similar here w/r/t underscores.

File: exporters/trace/jaeger/jaeger_test.go (hunk 2 above)
Comment (matej-g) [reply exporters/trace/jaeger/jaeger_test.go:49]: Honestly I'm used to both ways of test naming, even though I'm aware of the mixed caps convention. I have no strong feeling either way, so breaking it into sub-tests with `t.Run` is entirely fine with me.

File: exporters/trace/zipkin/zipkin.go (hunk 3 above)
Comment (matej-g) [reply exporters/trace/zipkin/zipkin.go:107]: Understood, in that case this was an oversight on my part. Keeping it as an argument makes perfect sense then.

File: exporters/trace/jaeger/jaeger_test.go (hunk 2 above)
Comment (lizthegrey) [reply exporters/trace/jaeger/jaeger_test.go:49]: please no underscores, thanks <3

Comment (matej-g) [reply exporters/trace/jaeger/jaeger_test.go]: @lizthegrey duly noted :wink:

**Comment by matej-g** [comment]
@MrAlias thanks for the feedback, I implemented the suggested changes

File: exporters/trace/zipkin/zipkin.go [hunk 5]
```diff
@@ -35,58 +38,105 @@ type Exporter struct {
...
+	o           options
 }
 
 var (
 	_ export.SpanBatcher = &Exporter{}
 )
 
 // Options contains configuration for the exporter.
-type Options struct {
+type options struct {
```
Comment (MrAlias) [inline exporters/trace/zipkin/zipkin.go:49]: Keeping this unexported sounds good. In https://github.com/open-telemetry/opentelemetry-go/issues/536 it seems like the consensus was to call this `Config`, but it can be for a separate PR (addressing https://github.com/open-telemetry/opentelemetry-go/issues/536 directly) if we don't want to address here.

**Comment by MrAlias** [comment]
Other than the binary purge, this is looking good and should be approved afterwards.

**Comment by matej-g** [comment]
> Other than the binary purge, this is looking good and should be approved afterwards. Great job

Sorry about that, removed the binary, hopefully now all is good :+1:
